mediator/src/controllers/cht.ts

```diff
@@ -21,6 +22,12 @@ export async function createPatient(chtPatientDoc: any) {
     chtPatientDoc.doc._id = await getPatientUUIDFromSourceId(chtPatientDoc.source_id);
   }
 
+  //check if patient already exists
+  const patient = await getFhirResourceByIdentifier(chtPatientDoc.doc.patient_id, 'Patient');
+  if (patient?.data?.total > 0){
+    return { status: 200, data: { message: `Patient with the same patient_id already exists`} };
```

Comment:
Could you explain in what scenario there would be multiple requests with the same patient_id, and we would want to prevent duplicate requests to updateFhirResource?

One thing is updates...which won't work throughout the entire flow anyway, but no reason to prevent them if we don't have to. The other thing is duplicate patients being created on CHT, which this won't prevent because they will still each get their own `patient_id`. Is there something else I'm missing?

Reply:
> Could you explain in what scenario there would be multiple requests with the same patient_id, and we would want to prevent duplicate requests to updateFhirResource?

One scenario would be if for some reason the post request to the `/patient` endpoint is retried manually from the OpenHIM dashboard even if it was successful.

Duplicates being sent to the CHT is handled [here](https://github.com/medic/cht-interoperability/pull/146/files#diff-edbc2a0030df5cda414a6fb0a6559af1dcc0452fd2a7d95b2e0c4ee9e9e7c580R139) by not calling `createChtPatient` if the request to `updateFhirResource` returns and update and not a new creation.